File: py/examples/simple_example.py
"""
简单示例：单个 Context 的 TTS 使用
"""

import asyncio
import os
from multi_ws_tts_sdk import TTSClient


async def main():
    # 从环境变量获取配置
    base_url = os.getenv("TTS_BASE_URL", "wss://your-domain.com")
    api_key = os.getenv("TTS_API_KEY")
    voice_id = os.getenv("TTS_VOICE_ID")
    
    if not api_key or not voice_id:
        print("❌ Please set TTS_API_KEY and TTS_VOICE_ID environment variables")
        return
    
    print("=== Multi-Context WebSocket TTS SDK - Simple Example ===\n")
    
    # 1. 创建客户端
    client = TTSClient(base_url, api_key, voice_id)
    
    # 2. 连接到服务器
    params = {
        "model_id": "flash_v2_5",
        "format": "pcm_16000",
        "language_code": "zh"
    }
    
    try:
        await client.connect(params)
        print("✅ 连接成功\n")
    except Exception as e:
        print(f"❌ 连接失败: {e}")
        return
    
    # 3. 创建 Context
    context = client.create_context("ctx_001")
    
    # 4. 设置回调
    total_audio_size = 0
    
    def on_audio(audio_data: bytes, is_final: bool):
        nonlocal total_audio_size
        total_audio_size += len(audio_data)
        final_text = " (最终)" if is_final else ""
        print(f"🎵 收到音频: {len(audio_data)} 字节, is_final={is_final}{final_text}, 累计: {total_audio_size} 字节")
    
    def on_error(error_code: str, error_message: str):
        print(f"❌ Context 错误: {error_code} - {error_message}")
    
    def on_complete():
        print("✅ Context 完成")
    
    context.on_audio(on_audio).on_error(on_error).on_complete(on_complete)
    
    # 5. 发送文本
    print("📤 发送文本: '你好，世界'")
    await context.send_text("你好，", flush=False)
    await context.send_text("世界", flush=True)
    
    # 6. 等待处理
    print("⏳ 等待 TTS 处理...\n")
    await asyncio.sleep(5)
    
    # 7. 发送 EOS
    print("📤 发送 EOS (结束输入)")
    await context.end_input()
    
    await asyncio.sleep(2)
    
    # 8. 获取所有音频
    all_audio = context.get_all_audio()
    print(f"\n📊 总音频大小: {len(all_audio)} 字节")
    
    # 可选：保存音频文件
    # with open("output.pcm", "wb") as f:
    #     f.write(all_audio)
    # print("💾 音频已保存到 output.pcm")
    
    # 9. 关闭
    await context.close()
    await client.disconnect()
    
    print("\n✅ 示例完成")


if __name__ == "__main__":
    asyncio.run(main())

File: java/examples/SimpleExample.java
import com.inccleo.tts.TTSClient;
import com.inccleo.tts.TTSContext;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

/**
 * 简单示例：单个 Context 的使用
 */
public class SimpleExample {
    public static void main(String[] args) {
        // 从环境变量读取配置
        String baseUrl = getEnv("TTS_BASE_URL", "ws://localhost:5001");
        String apiKey = getEnv("TTS_API_KEY", "your_api_key");
        String voiceId = getEnv("TTS_VOICE_ID", "your_voice_id");

        System.out.println("=== Multi-Context WebSocket TTS SDK - Simple Example ===\n");

        TTSClient client = null;
        try {
            // 1. 创建客户端
            client = new TTSClient(baseUrl, apiKey, voiceId);

            // 2. 连接到服务器
            Map<String, String> params = new HashMap<>();
            params.put("model_id", "flash_v2_5");
            params.put("format", "pcm_16000");
            params.put("language_code", "zh");

            System.out.println("🔌 连接到服务器...");
            client.connect(params);
            System.out.println("✅ 连接成功\n");

            // 3. 创建上下文
            TTSContext context = client.createContext("simple_context_001");
            System.out.println("📝 创建上下文: " + context.getContextId() + "\n");

            // 用于等待完成的 CountDownLatch
            CountDownLatch completeLatch = new CountDownLatch(1);

            // 4. 设置回调
            context.onAudio((audio, isFinal) -> {
                byte[] audioData = java.util.Base64.getDecoder().decode(audio);
                System.out.println("🎵 收到音频: " + audioData.length + " 字节" + 
                                 (isFinal ? " (最终帧)" : ""));
            })
            .onError((code, message) -> {
                System.err.println("❌ 错误: " + code + " - " + message);
                completeLatch.countDown();
            })
            .onComplete(() -> {
                System.out.println("\n✅ 上下文处理完成");
                completeLatch.countDown();
            });

            // 5. 发送文本
            String text = "你好，世界！这是一个测试。";
            System.out.println("📤 发送文本: '" + text + "'");
            context.sendText(text, true);

            // 6. 发送 EOS
            System.out.println("📤 发送 EOS (结束输入)\n");
            context.endInput();

            // 7. 等待处理完成（最多 10 秒）
            System.out.println("⏳ 等待 TTS 处理...\n");
            if (!completeLatch.await(10, TimeUnit.SECONDS)) {
                System.out.println("⚠️  等待超时");
            }

            // 8. 关闭上下文
            context.close();

            // 9. 显示统计信息
            System.out.println("\n============================================================");
            System.out.println("📊 统计信息:");
            System.out.println("============================================================");
            System.out.println("总音频块数: " + context.getAudioChunks().size());
            
            int totalBytes = 0;
            for (byte[] chunk : context.getAudioChunks()) {
                totalBytes += chunk.length;
            }
            System.out.println("总音频大小: " + totalBytes + " 字节");

            System.out.println("\n✅ 示例完成");

        } catch (Exception e) {
            System.err.println("❌ 错误: " + e.getMessage());
            e.printStackTrace();
        } finally {
            // 10. 断开连接
            if (client != null) {
                client.disconnect();
                System.out.println("🔌 已断开连接");
            }
        }
    }

    private static String getEnv(String name, String defaultValue) {
        String value = System.getenv(name);
        return (value != null && !value.isEmpty()) ? value : defaultValue;
    }
}
